flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesClusterDescriptor.java

```diff
@@ -108,17 +101,31 @@ public String getClusterDescription() {
 
 	@Override
 	public ClusterClientProvider<String> retrieve(String clusterId) {
-		return createClusterClientProvider(clusterId);
+		final ClusterClientProvider<String> clusterClientProvider = createClusterClientProvider(clusterId);
+
+		try (ClusterClient<String> clusterClient = clusterClientProvider.getClusterClient()) {
```

Comment:
I think the behavior is changed here. The previous way would not close the `clusterClient`, but now we close the client after logging. If so it should not be a hotfix commit and might need a unit test to cover the change.

Reply:
We do not change any behavior here. We just get the real cluster client of `ClusterClientProvider` for logging and then close it. The `ClusterClientProvider` do not need to be close, the `ClusterClientProvider.getClusterClient()` should do.

Reply:
I was also referring to the `ClusterClientProvider.getClusterClient()`. Before the changes, it is out of `try` clause, so it would not be closed after logging. But now it would be closed after logging.

Reply:
Before changes, `createClusterClientProvider` is called in `deployClusterInternal` and put into `try (ClusterClient<String> clusterClient = clusterClientProvider.getClusterClient()) {}`.

I just move the log to another place.